algoliasearch-client-js/algoliasearch-client-js.d.ts

```diff
@@ -0,0 +1,1563 @@
+     */
+    deleteIndex(name:string, cb:Function) : void;
+    /**
+     * Delete a specific index
+     * @param name
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#delete-index---deleteindex
+     */
+    deleteIndex(name:string) : Promise<any>;
+    /**
+     * Copy an  index from a specific index to a new one
+     * @param from origin index
+     * @param to destination index
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#copy-index---copyindex
+     */
+    copyIndex(from:string, to:string, cb:Function) : void;
+    /**
+     * Copy an  index from a specific index to a new one
+     * @param from origin index
+     * @param to destination index
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#copy-index---copyindex
+     */
+    copyIndex(from:string, to:string) : Promise<any>;
+    /**
+     * Move index to a new one (and will overwrite the original one)
+     * @param from origin index
+     * @param to destination index
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#move-index---moveindex
+     */
+    moveIndex(from:string, to:string, cb:Function) : void;
+    /**
+     * Move index to a new one (and will overwrite the original one)
+     * @param from origin index
+     * @param to destination index
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#move-index---moveindex
+     */
+    moveIndex(from:string, to:string) : Promise<any>;
+    /**
+     * Generate a public API key
+     * @param key api key
+     * @param filters
+     * https://github.com/algolia/algoliasearch-client-js#generate-key---generatesecuredapikey
+     */
+    generateSecuredApiKey(key:string, filters:IAlgoliaSecuredApiOptions) : void;
+    /**
+     * Perform multiple operations with one API call to reduce latency
+     * @param action
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#custom-batch---batch
+     */
+    batch(action:IAlgoliaAction, cb:Function) : void;
+    /**
+     * Perform multiple operations with one API call to reduce latency
+     * @param action
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#custom-batch---batch
+     */
+    batch(action:IAlgoliaAction) : Promise<any>;
+    /**
+     * Browse an index
+     * @param query
+     * @param cb(err, content)
+     * https://github.com/algolia/algoliasearch-client-js#backup--export-an-index---browse
+     */
+    browse(query:string, cb:Function) : void;
+    /**
+     * Browse an index
+     * @param query
+     * return {Promise}
+     * https://github.com/algolia/algoliasearch-client-js#backup--export-an-index---browse
+     */
+    browse(query:string) : Promise<{cursor? : string, hits: any[], params: string, query: string, processingTimeMS: number}>;
```

Comment:
Since the type inside the promise is used more than once, define an interface for it.